src/main/drivers/pwm_output.c

```diff
@@ -129,18 +130,73 @@ static void pwmWriteMultiShot(uint8_t index, uint16_t value)
     *motors[index].ccr = lrintf(((float)(value-1000) * MULTISHOT_20US_MULT) + MULTISHOT_5US_PW);
 }
 
-void pwmWriteMotor(uint8_t index, uint16_t value)
+#ifdef USE_DSHOT
+static void pwmWriteDigital(uint8_t index, uint16_t value)
+{
+    pwmMotorOutput_t * const motor = &motors[index];
+
+    motor->value = value;
+    motor->updateFlags |= MOTOR_UPDATE_VALUE;
+}
+
+void pwmCompleteDigitalMotorUpdate(uint8_t motorCount)
+{
+    UNUSED(motorCount);
+
+}
+
+void pwmWriteValueToDmaBuffer(uint16_t value, uint32_t *buffer, uint8_t flags)
+{
+    buffer[0]  = (value & 0x400) ? MOTOR_BIT_1 : MOTOR_BIT_0;
```

Comment:
ok yeah right. Your above suggested code is nice btw!

The current CRC doesnt do the job. Thats why my testing was failing.
(value & 0x400) ^ (value & 0x40) returns integer. Should rather be (bool)(value & 0x400) ^ (bool)(value & 0x40)

But above const uint16_t csum = (value ^ (value >> 4) ^ (value >> 8)) & 0xf;  is nice. Does take more time to understand though